### PR TITLE
KTOR-6741 Part 1 - Replace "Creating HTTP API's tutorial" with a new tutorial "Handle requests and generate responses"

### PR DESCRIPTION
**Pull request by vnikolova**
[KTOR-6741](https://youtrack.jetbrains.com/issue/KTOR-6741/Replace-Creating-HTTP-APIs-tutorial-with-new-content)

The "Creating HTTP APIs" tutorial will be replaced by two new tutorials:

1. Handle requests and generate responses. (orig. [Understanding Routing and Requests](https://docs.google.com/document/d/10XH3fwAG8GY1s6BydU8tVH0Dtxek29sPuwavYfT3Tk0/edit?usp=sharing))
2. Create RESTful APIs that generate JSON. (orig. [Content Negotiation and REST](https://docs.google.com/document/d/1Ekah3oGHz50wO2pis65l1hQllHXKfX7uQSUWOK9uXKc/edit?usp=sharing))

In this PR:
[See preview on staging](https://ktor.shared.aws.intellij.net/docs/handle-requests-and-generate-responses.html)
- [x] Both new topics are added and placed in the navigation
- [x]  "Creating HTTP APIs" is removed
- [x]  Content is added for "Handle requests and generate responses" tutorial with screenshots for light and dark theme
- [x]  An example project is added

The "Create RESTful APIs that generate JSON." topic is currently empty. It will be covered in a separate PR, to be merged to `main` later with this one.

### CODE REVIEW

File: codeSnippets/snippets/tutorial-server-routing-and-requests/src/main/kotlin/model/TaskRepository.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+object TaskRepository {
+    private val tasks = mutableListOf(
+        Task("cleaning", "Clean the house", Priority.Low),
+        Task("gardening", "Mow the lawn", Priority.Medium),
+        Task("shopping", "Buy the groceries", Priority.High),
+        Task("painting", "Paint the fence", Priority.Medium)
+    )
+
+    fun allTasks(): List<Task> = tasks
+    fun tasksByPriority(priority: Priority) = tasks.filter {
```
Comment (e5l) [inline codeSnippets/snippets/tutorial-server-routing-and-requests/src/main/kotlin/model/TaskRepository.kt:12]: I would keep an empty line between methods for readability; you can invoke autoformat in IDEA or use detekt plugin, they can fix it

File: topics/handle-requests-and-generate-responses.topic [hunk 2]
```diff
@@ -0,0 +1,951 @@
...
+            <li>Use the repository to find all the tasks with the specified priority.</li>
+            <li>If there are no matching tasks return a <code>404</code> status (Not Found).</li>
+            <li>Return the matching tasks formatted as an HTML table.</li>
+        </list>
+        <p>
+            You will first implement this functionality, and then find the best way to check it is working.
+        </p>
+        <procedure title="Add a new route">
+            <step>
+                <p>Open the
```
Comment (e5l) [inline topics/handle-requests-and-generate-responses.topic:410]: image

code blocks merged

**Comment by e5l** [comment]
some screenshots are flying :) 
image

**Review by e5l — CHANGES_REQUESTED** [review]
Good job, it looks really nice! Please check some minor comments.

I also noticed that screenshots are only for the dark theme now

File: topics/handle-requests-and-generate-responses.topic [hunk 3]
```diff
@@ -0,0 +1,951 @@
...
+            <li>View tasks by priority and name, again as HTML.</li>
+            <li>Add additional tasks by submitting an HTML form.</li>
+        </list>
+        <p>
+            You will do the minimum possible to get some basic functionality working, and then improve and extend this
+            functionality over seven iterations. This minimum functionality will consist of a project containing some
+            model types, a list of values and a single route.
+        </p>
+    </chapter>
+    <chapter title="Iteration 1: Display static HTML content" id="iteration-1">
```
[the remark this answers is not in the document]
Comment (vnikolova) [reply topics/handle-requests-and-generate-responses.topic:57]: Or, what do you think about removing "Iteration {n}:"  from the titles altogether? I don't see it as a common practice to have either one in the title.

File: topics/handle-requests-and-generate-responses.topic [hunk 4]
```diff
@@ -0,0 +1,951 @@
...
+            <li><a anchor="create-sample-values">Declare a list of tasks containing sample values.</a></li>
+            <li><a anchor="add-a-route">Modify the route and request handler to return this list.</a></li>
+            <li><a anchor="test">Test that the new feature works using the browser.</a></li>
+        </list>
+        <procedure title="Create model types" id="create-model-types">
+            <step>
+                <p>In the
+                    <ui-path>Project</ui-path>
+                    tool window, right-click the
+                    <path>src/main/kotlin/com.example</path>
```
[the remark this answers is not in the document]
Comment (vnikolova) [reply topics/handle-requests-and-generate-responses.topic:146]: Hm I notice now that IntelliJ displays `com.example` when the view is __Project__ and `com/example` when it is __Project Files__. Maybe we can replace these instructions with more generic ones? Or do we want to include intelliJ-specific instructions as well?

For example: 
"Inside `src/main/kotlin/com/example` create a new directory called "model"."

[the remark this answers is not in the document]
Comment (e5l) [reply topics/handle-requests-and-generate-responses.topic:146]: lgtm

File: topics/handle-requests-and-generate-responses.topic (hunk 3 above)
[the remark this answers is not in the document]
Comment (e5l) [reply topics/handle-requests-and-generate-responses.topic:57]: yep, sounds good

**Pull request merged**